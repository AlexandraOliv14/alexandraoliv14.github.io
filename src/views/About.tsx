import React, { useContext, useEffect, useRef, useState } from 'react'
import { FiGithub, FiLinkedin, FiArrowUp, FiArrowDown } from "react-icons/fi";

import { Separacion } from '../components/Separacion'
import { ButtonIcon } from '../components/ButtonIcon'
import { ThemeContext } from '../context/themeContext';
import { theme, size } from '../theme';
import Typewriter from "typewriter-effect";
import useInViewPort from '../hooks/useInViewPort';
import FadeInText from '../components/FadeInText';

export const About = () => {
    
    const { toggleState, currentState } = useContext(ThemeContext);
    const [down, setDown] = useState(true);

    const targetRef = useRef<HTMLDivElement>(null);
    const {inViewport} = useInViewPort(targetRef, { threshold: 0.5 });

    const aboutEduRef = useRef<HTMLDivElement>(null);
    const {observed:inViewportEdu } = useInViewPort(aboutEduRef, { threshold: 0.5 });

    const findRef = useRef<HTMLDivElement>(null);
    const {inViewport:inViewportFind} = useInViewPort(findRef, { threshold: 0.5 });

    const aboutRef = useRef<HTMLDivElement>(null);
    const {inViewport: inViewAbout} = useInViewPort(aboutRef, { threshold: 0.5 });

    const footerRef = useRef<HTMLDivElement>(null);
    const {inViewport: inViewFooter} = useInViewPort(footerRef, { threshold: 0.5 });

    useEffect(() => {
        toggleState('ligthwarm')
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    useEffect(() => {
        if (inViewport || inViewAbout) {
            setDown(true)
        }
        if (inViewportEdu || inViewFooter) {
            setDown(false)
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [inViewport, inViewportFind])



    return (
        <div className='about'>
            <div className='about-me' ref={aboutRef}>
                <ButtonIcon href={down ? '#about-findme' : '#about'} color={theme[currentState].primary} fixed={true}>
                    {
                        down ?
                            <FiArrowDown style={{ fontSize: size['small'], color: 'black' }} /> :
                            <FiArrowUp style={{ fontSize: size['small'], color: 'black' }} />
                    }
                </ButtonIcon>
                <div className='img'>
                    <img src='./imgs/aleyPatana.jpg' width={417} height={417} alt='aleypatana' />
                </div>
                <div className='p'>
                    <Typewriter
                        options={{ delay: 30 }}
                        onInit={(typewriter) => {
                            typewriter
                                .typeString('¬°Hola! Soy Alexandra Olivares<br /> Sol√≠s, Ingeniera inform√°tica<br />especializada en desarrollo<br />de software full-stack, con <br />un enfoque en React.js, React<br />Native, y Java.')
                                .start();
                        }}
                    />
                </div>
            </div>

            <Separacion />
            <div className='about-educacion'>
                <p ref={targetRef} className={inViewport ? 'animacion' : 'hidden'}>
                    Me gradu√© de la<br />
                    Universidad Tecnol√≥gica<br />
                    Metropolitana y <br />
                    actualmente estoy <br />
                    expandiendo mis <br />
                    conocimientos a trav√©s <br />
                    de un m√°ster en <br />
                    Ingenier√≠a Inform√°tica <br />
                    en la Universidad <br />
                    Polit√©cnica de Catalu√±a.
                </p>
                <div className='img'>
                    <img src='./imgs/DALL_E_diploma.png' width={417} height={417} alt='dallydiploma' style={{ background: theme[currentState].primary }} />
                </div>
            </div>
            <Separacion />
            <div className='about-actual'>
                <div className='img'>
                    <img src='./imgs/DALL_E_crecimiento.png' width={417} height={417} alt='dallydiploma' style={{ background: theme[currentState].primary }} />
                </div>
                <div className='container-text' ref={aboutEduRef}>
                    {inViewportEdu ? <FadeInText text="Me apasiona profundamente  trabajar en proyectos      innovadores y me     encuentro constantemente   en b√∫squeda de nuevas       oportunidades y     aprendizajes en el mundo del desarrollo." />
                        : <FadeInText text='   ' />}
                </div>
                <div className='container-text_min'>
                    <p>Me apasiona profundamente trabajar en proyectos innovadores y me encuentro constantemente en b√∫squeda de nuevas oportunidades y aprendizajes en el mundo del desarrollo.</p>
                </div>
            </div>
            <Separacion />
            <div className='about-findme' ref={findRef} id='about-findme'>
                {inViewportFind && <h1>üîé
                    <Typewriter
                        onInit={(typewriter) => {
                            typewriter
                                .typeString(' Puedes encontrarme en')
                                .pauseFor(1000)
                                .start();
                        }}
                    />
                </h1>}
                <div className='links' id='link' ref={footerRef}>
                    <ButtonIcon href='https://www.linkedin.com/in/alexandraos14/' color={theme[currentState].primary}>
                        <FiLinkedin style={{ fontSize: size['large'], color: 'black' }} />
                    </ButtonIcon>
                    <ButtonIcon href='https://github.com/AlexandraOliv14' color={theme[currentState].primary}>
                        <FiGithub style={{ fontSize: size['large'], color: 'black' }} />
                    </ButtonIcon>
                </div>
            </div>
        </div>
    )
}