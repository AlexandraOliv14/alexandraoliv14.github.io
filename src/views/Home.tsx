import React from 'react'
import { FiLinkedin, FiGithub  } from "react-icons/fi";

export const Home = () => {

    return (
        <div className='containerHome'>
            <div className='containerInfo'>
                <h1>Hola, soy Alexandra Olivares <br/>
                </h1>
                <p>
                Una  desarrolladora de software Full Stack 
                con una pasión por crear aplicaciones web y 
                móviles. He trabajando con tecnologias como
                React, React Native, Java, Python entre otras.
                </p>
                <br/>
            </div>
            <div className='links'>
                <a href='https://www.linkedin.com/in/alexandra-olivares-solis-3950081b5' target="_blank" rel="noopener noreferrer">
                    <FiLinkedin style={{fontSize:40, margin:'5px'}}/>
                </a>
                <a href='https://github.com/AlexandraOliv14' target="_blank" rel="noopener noreferrer">
                    <FiGithub style={{fontSize:40, margin:'5px'}}/>
                </a>
            </div>
            
        </div>
    )
}