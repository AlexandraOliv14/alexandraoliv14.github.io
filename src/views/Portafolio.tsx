import React, { useContext, useEffect } from 'react'
import { ThemeContext } from '../context/themeContext';
import { size, sizeType, theme } from '../theme';
import { Button, ButtonIcon, Select, Input, InputColor } from '../components';
import { useForm } from '../hooks';
import { FiGithub } from 'react-icons/fi';

export const Portafolio = () => {

    const { toggleState, currentState } = useContext(ThemeContext);//currentState

    useEffect(() => {
        toggleState('warm')
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    const { state, onChange, setState } = useForm({ select: 'warm', text: 'Home', color: theme[currentState].primary, color1: theme[currentState].primary, sizeButton: 'medium' as sizeType});
    const { select, text, color, color1, sizeButton} = state;

    useEffect(() => {
        toggleState(select);

        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [select]);

    useEffect(() => {
        setState({ ...state, color: theme[currentState].primary, color1: theme[currentState].primary })
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [currentState])


    return (
        <div className='portafolio'>
            <div className='pagina1'>
                <div className='card' style={{ background: theme[currentState].secondary }}>
                    <div className='card-flex'>
                        <h2>Componentes de este proyecto</h2>
                        <div className='theme'>
                            <h3>Manejo de tema</h3>
                            <div className='select-theme'>
                                <Select label='Selecciona el tema' name='select' onChange={onChange} value={select} opcions={['warm', 'ligthwarm', 'cool']} />
                            </div>
                        </div>
                        <div className='theme-button'>
                            <h3>Manejo de botones</h3>
                            <div className='button'>
                                <div className='button-item'>
                                    <Button text={text} color={color} />
                                </div>
                                <div className='button-item'>
                                    <Input label='Escriba el nombre del bot칩n: ' name='text' onChange={onChange} value={text} />
                                </div>
                                <div className='button-item'>
                                    <InputColor label='Seleccione el color del bot칩n:' name='color' onChange={onChange} value={color} />
                                </div>
                            </div>
                            <div className="button2">
                                <div className='button-item'>
                                    <ButtonIcon color={color1}>
                                        <FiGithub style={{ fontSize: size[sizeButton as sizeType], color: 'black' }} />
                                    </ButtonIcon>
                                </div>
                                <div className='button-item'>
                                <Select label='Selecciona el tama침o: ' name='sizeButton' onChange={onChange} value={sizeButton} opcions={['small', 'medium', 'large']} />
                                </div>
                                <div className='button-item'>
                                    <InputColor label='Seleccione el color del bot칩n:' name='color1' onChange={onChange} value={color1} />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
