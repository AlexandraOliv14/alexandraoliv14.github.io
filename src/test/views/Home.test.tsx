import { render, screen } from "@testing-library/react"
import { Home } from "../../views/Home"

describe('Pruebas en <Home/>', () => {
    test('Bienvenida',()=>{
        render(<Home/>);
        const elemnt = screen.getByText(/Soy Alexandra Olivares/i);
        expect(elemnt).toBeInTheDocument();
    })

    test('Debe coincidir con Snapshot',()=>{
        const {container} = render(<Home/>);
        expect(container).toMatchSnapshot();
    })

    test('Debe contener foto animada',()=>{
        render(<Home/>);
        const img = screen.getByAltText('aleAnimada')
        expect(img).toBeTruthy();
    })

    test('Debe contener foto animada con url correspondiente',()=>{
        render(<Home/>);
        const img = screen.getByAltText('aleAnimada')
        expect(img).toHaveAttribute('src', './imgs/aleAnimada.png')
    })
})
