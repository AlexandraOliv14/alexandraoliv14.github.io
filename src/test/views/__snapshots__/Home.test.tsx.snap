// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Home/> Debe coincidir con Snapshot 1`] = `
<div>
  <div
    class="containerHome"
  >
    <div
      class="containerFoto"
    >
      <img
        alt="aleAnimada"
        src="./imgs/aleAnimada.png"
        width="300"
      />
    </div>
    <div
      class="containerInfo"
    >
      <div
        class="saludo"
      >
        <h1>
          Hola ðŸ‘‹
          <br />
          Soy Alexandra Olivares
        </h1>
      </div>
      <p>
        Una programadora full-stack 
        <br />
        con amplia experiencia en
        <br />
        React.js, React Native y Java.
      </p>
    </div>
  </div>
</div>
`;
